staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -229,11 +229,11 @@ func (config *DirectClientConfig) getUserIdentificationPartialConfig(configAuthI
 	if len(configAuthInfo.Token) > 0 {
 		mergedConfig.BearerToken = configAuthInfo.Token
 	} else if len(configAuthInfo.TokenFile) > 0 {
-		tokenBytes, err := ioutil.ReadFile(configAuthInfo.TokenFile)
-		if err != nil {
+		ts := restclient.NewCachedFileTokenSource(configAuthInfo.TokenFile)
```

Comment:
This broke clients that depend on these methods.  We shouldn't have broken this without providing an alternate path.

Reply:
I agree with Clayton here, `BearerToken` was never deprecated and we don't provide alternative, yet suddenly it's empty. We should deprecate this field and provide users with explanation what they should be using instead.

Reply:
> The transformation of TokenFile -> bearer token (and what InClusterConfig returns) is what is in question.

Yes, sorry for my short-circuited answer :)

Reply:
I'm looking into plumbing both BearerToken and TokenFile out to the config and moving the refresh behavior into the round tripper constructed from the config

Reply:
opened https://github.com/kubernetes/kubernetes/pull/71713